dipy/segment/bundles.py

```diff
@@ -0,0 +1,406 @@
+                select_target=slr_select[1],
+                method=slr_method)
+        else:
+            transf_streamlines = neighb_streamlines
+
+        pruned_streamlines, labels = self._prune_what_not_in_model(
+            model_centroids,
+            transf_streamlines,
+            neighb_indices,
+            pruning_thr=pruning_thr,
+            pruning_distance=pruning_distance)
+
+        if self.verbose:
+            print('Total duration of recognition time is %0.3f sec.\n'
+                  % (time()-t,))
+        # return recognized bundle in original streamlines, labels of
+        # recognized bundle and transformed recognized bundle
+        return pruned_streamlines, labels, self.streamlines[labels]
+
+    def _cluster_model_bundle(self, model_bundle, model_clust_thr, nb_pts=20,
+                              select_randomly=500000):
+
+        if self.verbose:
+            t = time()
+            print('# Cluster model bundle using QBX')
+            print(' Model bundle has %d streamlines'
+                  % (len(model_bundle), ))
+            print(' Distance threshold %0.3f' % (model_clust_thr,))
+        thresholds = self.start_thr + [model_clust_thr]
+
+        model_cluster_map = qbx_and_merge(model_bundle, thresholds,
+                                          nb_pts=nb_pts,
+                                          select_randomly=select_randomly,
+                                          rng=None,
+                                          verbose=self.verbose)
+        model_centroids = model_cluster_map.centroids
+        nb_model_centroids = len(model_centroids)
+
+        if self.verbose:
+            print(' Model bundle has %d centroids'
+                  % (nb_model_centroids,))
+            print(' Duration %0.3f sec. \n' % (time() - t, ))
+        return model_centroids
+
+    def _reduce_search_space(self, model_centroids,
+                             reduction_thr=20, reduction_distance='mdf'):
+        if self.verbose:
+            t = time()
+            print('# Reduce search space')
+            print(' Reduction threshold %0.3f' % (reduction_thr,))
+            print(' Reduction distance {}'.format(reduction_distance))
+
+        if reduction_distance.lower() == 'mdf':
+            if self.verbose:
+                print(' Using MDF')
+            centroid_matrix = bundles_distances_mdf(model_centroids,
+                                                    self.centroids)
+        elif reduction_distance.lower() == 'mam':
+            if self.verbose:
+                print(' Using MAM')
+            centroid_matrix = bundles_distances_mdf(model_centroids,
+                                                    self.centroids)
+        else:
+            raise ValueError('Given reduction distance not known')
+
+        centroid_matrix[centroid_matrix > reduction_thr] = np.inf
+
+        mins = np.min(centroid_matrix, axis=0)
+        close_clusters_indices = list(np.where(mins != np.inf)[0])
```

Comment:
I don't get this strategy. Why not forget the infinity and simply use your `reduction_thr` directly in the `np.where`?